poly/helper_test.go

```diff
@@ -0,0 +1,106 @@
+package poly
+
+import (
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"github.com/dedis/crypto/random"
+)
+
+var edward = SUITE
+
+func generateKeyPair() *config.KeyPair {
+	keypair := new(config.KeyPair)
+	keypair.Gen(edward, random.Stream)
+	return keypair
+}
+
+func generateKeyPairList(n int) []*config.KeyPair {
+	l := make([]*config.KeyPair, n)
+	for i := 0; i < n; i++ {
+		l[i] = generateKeyPair()
+	}
+	return l
+}
+
+func generatePublicListFromPrivate(private []*config.KeyPair) []abstract.Point {
+	l := make([]abstract.Point, len(private))
+	for i := 0; i < len(private); i++ {
+		l[i] = private[i].Public
+	}
+	return l
+}
+
+// produce M receivers from their private/pub keys
+func generateReceivers(info PolyInfo, keys []*config.KeyPair) []*Receiver {
+	n := len(keys)
+	l := make([]*Receiver, n)
+	for i := 0; i < n; i++ {
+		l[i] = NewReceiver(info, keys[i])
+	}
+	return l
+}
+
+// Produce N dealers with the public keys of the M receivers
+func generateDealers(n int, info PolyInfo, receiverList []abstract.Point) []*Dealer {
+	d := make([]*Dealer, n)
+	for i := 0; i < n; i++ {
+		d[i] = NewDealer(info, generateKeyPair(), generateKeyPair(), receiverList)
+	}
+	return d
+}
+
+// Returns N dealers with M receivers with the right keys / public keys ...
+func generateNDealerMReceiver(info PolyInfo, n, m int) ([]*Dealer, []*Receiver) {
+	receiverKeys := generateKeyPairList(m)
```

Comment:
Are all the above four functions called only from here?  Looks like it, and if so, having all those separate functions just seems to be creating unnecessary code-bloat.  Just inline the four functions here.  And while you're at it, just have one loop to initialize all four arrays together.  The code will be way more concise and simpler.